SHMEM/isx.c

```diff
@@ -722,106 +731,62 @@ static void print_run_info(FILE * fp)
 
 /*
  * Prints all of the timining information for an individual PE as a row
- * to the file specificed by 'fp'. 
+ * to the file specificed by 'fp'.
  */
 static void print_timer_values(FILE * fp)
 {
-  unsigned int num_records = NUM_PES * NUM_ITERATIONS; 
-
-  for(uint64_t i = 0; i < num_records; ++i) {
+  for(uint64_t i = 0; i < NUM_ITERATIONS; ++i) {
     for(int t = 0; t < TIMER_NTIMERS; ++t){
-      if(timers[t].all_times != NULL){
-        fprintf(fp,"%f\t", timers[t].all_times[i]);
+      if(timers[t].seconds_iter > 0){
+        fprintf(fp,"%f\t", timers[t].seconds[i]);
       }
-      if(timers[t].all_counts != NULL){
-        fprintf(fp,"%u\t", timers[t].all_counts[i]);
+      if(timers[t].count_iter > 0){
+        fprintf(fp,"%u\t", timers[t].count[i]);
       }
     }
     fprintf(fp,"\n");
   }
 }
 
-/* 
+/*
  * Aggregates the per PE timing information
- */ 
+ */
 static double * gather_rank_times(_timer_t * const timer)
 {
   if(timer->seconds_iter > 0) {
 
     assert(timer->seconds_iter == timer->num_iters);
 
-    const unsigned int num_records = NUM_PES * timer->seconds_iter;
-#ifdef OPENSHMEM_COMPLIANT
-    double * my_times = shmem_malloc(timer->seconds_iter * sizeof(double));
-#else
-    double * my_times = shmalloc(timer->seconds_iter * sizeof(double));
-#endif
-    memcpy(my_times, timer->seconds, timer->seconds_iter * sizeof(double));
-
-#ifdef OPENSHMEM_COMPLIANT
-    double * all_times = shmem_malloc( num_records * sizeof(double));
-#else
-    double * all_times = shmalloc( num_records * sizeof(double));
-#endif
-
-    shmem_barrier_all();
-
-    shmem_fcollect64(all_times, my_times, timer->seconds_iter, 0, 0, NUM_PES, pSync);
     shmem_barrier_all();
 
 #ifdef OPENSHMEM_COMPLIANT
-    shmem_free(my_times);
+    double * my_average = shmem_malloc(sizeof(double));
 #else
-    shfree(my_times);
+    double * my_average = shmalloc(sizeof(double));
 #endif
+    double temp = 0.0;
+    for(int i = 0; i < timer->seconds_iter; i++) {
```

Comment:
I suppose you should get rid of the compiler error with something like this:
```suggestion
    for(unsigned int i = 0; i < timer->seconds_iter; i++) {
```